File: Data Structures & Algorithms/three-integer-sum/submission-2.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        vector<vector<int>> out = {};


        std::sort(nums.begin(), nums.end());
        int index = 0;
        while (index < nums.size()) {

            int val = nums[index];
            int l=index+1, r=nums.size() - 1;

            while (l < r) {
                if (l == index) {
                    l++; continue;
                }

                if (r == index) {
                    r--; continue;
                }

                int sum = val + nums[l] + nums[r];
                if (sum > 0) {
                    r--; 
                } else if (sum < 0) {
                    l++;
                } else {
                    out.push_back({val, nums[l], nums[r]});
                    r--;
                    l++;

                    while (l < r && nums[l] == nums[l-1]) l++;
                }
            }

            while (index < nums.size() && nums[index] == val)
                index++; 
        }

        return out;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        heap = [] 
        res = []

        nums.sort()

        count = 0
        curr = nums[0]

        for i in nums:
            if curr != i:
                heapq.heappush(heap, (-count, curr))
                curr = i 
                count = 1
            else:
                count += 1         
        heapq.heappush(heap, (-count,curr))

        while k != 0:
            count, val = heapq.heappop(heap)
            res.append(val)
            k -= 1
            
        return res
